src/why3prop.ml

```diff
@@ -0,0 +1,122 @@
+  ; symb_bot   : sym * pp_hint (** Bot(⊥) symbol.                   *)
+  ; symb_not   : sym * pp_hint (** Not(¬) symbol.                   *) }
+
+(** [get_prop_config builtins] set the builtins configuration using
+    [builtins] *)
+let get_prop_config : Proof.builtins -> prop_config = fun builtins ->
+  let find_sym key =
+    try StrMap.find key builtins with Not_found ->
+     Console.fatal_no_pos "Builtin symbol [%s] undefined." key
+  in
+  { symb_P     = find_sym "P"
+  ; symb_T     = find_sym "T"
+  ; symb_or    = find_sym "or"
+  ; symb_and   = find_sym "and"
+  ; symb_imp   = find_sym "imp"
+  ; symb_bot   = find_sym "bot"
+  ; symb_not   = find_sym "not" }
+
+(** [translate builtins (hs, g)] translate from lambdapi to Why3 goal [g]
+    using the hypothesis [hs]. *)
+let rec translate : Proof.builtins -> (Env.env * term) ->
+    cnst_table * (string * Why3.Term.term) list * Why3.Term.term =
+    fun builtins  (hs, g) ->
+    let cfg = get_prop_config builtins in
+    let (l_prop, hypothesis) =
+        List.fold_left (translate_context cfg) ([], []) hs in
+    try
+        let (l_prop, formula) = t_goal cfg l_prop g in
+        (l_prop, hypothesis, formula)
+    with NoGoalTranslation ->
+        Console.fatal_no_pos "The term [%a] is not of the forme [P _]"
+        Print.pp g
+
+(** [translate_context cfg (l_constants, l_hypothesis) (hyp_name, (_, hyp))]
+    translate the context [hyp] with the label [hyp_name] and add it in
+    [l_hypothesis] with the why3 constants [l_constants]. *)
+and translate_context :
+    prop_config ->
+    cnst_table * (string * Why3.Term.term) list ->
+    string * (tvar * tbox) ->
+    cnst_table * (string * Why3.Term.term) list =
+    fun cfg (l_constants, l_hypothesis) (hyp_name, (_, hyp)) ->
+    try
+        let (new_why3_l, hyp') = t_goal cfg l_constants (Bindlib.unbox hyp) in
+            (new_why3_l, (hyp_name, hyp')::l_hypothesis)
+    with NoGoalTranslation ->
+            (l_constants, l_hypothesis)
+
+(** [t_goal cfg l_prop trm] translate the lambdapi term [trm] to Why3 term
+    using the configuration [cfg] and the list of Why3 constants in [l_prop].
+    *)
+and t_goal : prop_config -> cnst_table -> term ->
+    cnst_table * Why3.Term.term =
+    fun cfg l_prop trm ->
+    match Basics.get_args trm with
+    | (symbol, [t]) when Basics.is_symb (fst cfg.symb_P) symbol ->
+        t_prop cfg l_prop [] t
+    | _                                                         ->
+        raise NoGoalTranslation
+
+(** [t_prop cfg l_prop ctxt p] translate the term [p] into Why3 terms with a
+    context [ctxt] and a config [cfg]. *)
+and t_prop :
+    prop_config -> cnst_table -> Ctxt.t -> term ->
+    cnst_table * Why3.Term.term =
+    fun cfg l_prop ctxt p ->
+    let (s, args) = Basics.get_args p in
+    match s with
+    | symbol when Basics.is_symb (fst cfg.symb_and) symbol  ->
+        let (l_prop, t1) = t_prop cfg l_prop ctxt (List.nth args 0) in
```

Comment:
List.nth can fail here!... You should instead do:
```
match s, args with
| s, [t1; t2] when ... ->
```

Reply:
Fixed in 64670391ee3fe852548ebaa1640358bd87899c5b